### ISSUE TITLE
Issues creating process tree with custom schema

### ISSUE THREAD

**Issue by jaimeatwork** [microsoft/msticpy#203]
**Describe the bug**
I am trying to create a schema from some macOS logs created by the Elastic agent. I have attempted to implement a custom schema as described [here](https://msticpy.readthedocs.io/en/latest/visualization/ProcessTree.html#adapting-the-input-schema-of-your-data). I am getting an error 

`UnboundLocalError: local variable 'extr_proc_tree' referenced before assignment`

that does not appear to be any variable I have control over and am not sure if this is a bug or an error in how I am using this. 

**To Reproduce**

1. Clone the repo here - https://github.com/jaimeatwork/macOS-ATTACK-DATASET
2. Use a jupyter notebook and virtualenv as described in your installation instructions
3. Read the data with the following code 

```python
import json
import pandas as pd
import yaml
import glob
from msticpy.sectools import ptree
from msticpy.sectools.proc_tree_builder import build_process_tree
from msticpy.nbtools import process_tree 

# Assumes this path is correct, may need to adjust depending on where you cloned the repo
files = glob.glob("./**/*.json")
attacks = []
for file in files:
    with open(file, 'r') as f:
        attack = {}
        data = json.loads(f.read())
        attack[file] = data
        attacks.append(attack)

# picked a single file with 10 events to build a sample tree
for attack in attacks:
    for k, v in attack.items():
        if k == "./Persistence/persistence_emond_process_execution.json":
            df = pd.json_normalize(v['documents'])



from msticpy.sectools.proc_tree_builder import build_process_tree, LX_EVENT_SCH
from msticpy.sectools import proc_tree_build_winlx as winlx
from copy import copy

cust_lx_schema = copy(LX_EVENT_SCH)
cust_lx_schema.time_stamp = "_source.@timestamp"
cust_lx_schema.host_name_column = "_source.host.hostname"
cust_lx_schema.process_id = "_source.process.pid"
cust_lx_schema.parent_id = "_source.process.parent.pid"
cust_lx_schema.cmd_line = "_source.process.command_line"
cust_lx_schema.event_id_column = "_source.data_stream.dataset"
cust_lx_schema.cmd_line = "_source.process.command_line"
cust_lx_schema.event_id_identifier = "endpoint.events.process"
cust_lx_schema.path_separator = "/"

process_tree.plot_process_tree(df, schema=cust_lx_schema)
```

4. Observe the error

error output 
```

---------------------------------------------------------------------------
UnboundLocalError                         Traceback (most recent call last)
/var/folders/6j/kg1frr2j61l2_xx9m01j_x3w0000gn/T/ipykernel_64665/961226578.py in <module>
     13 cust_lx_schema.path_separator = "/"
     14 
---> 15 process_tree.plot_process_tree(df, schema=cust_lx_schema)

~/.local/share/virtualenvs/macOS-ATTACK-DATASET-JrS6VCr6/lib/python3.8/site-packages/msticpy/nbtools/process_tree.py in plot_process_tree(data, schema, output_var, legend_col, show_table, **kwargs)
    187     pid_fmt = kwargs.pop("pid_fmt", "hex")
    188 
--> 189     data, schema, levels, n_rows = _pre_process_tree(data, schema, pid_fmt=pid_fmt)
    190     if schema is None:
    191         raise ProcessTreeSchemaException("Could not infer data schema from data set.")

~/.local/share/virtualenvs/macOS-ATTACK-DATASET-JrS6VCr6/lib/python3.8/site-packages/msticpy/nbtools/process_tree.py in _pre_process_tree(proc_tree, schema, pid_fmt)
    298     missing_cols = _check_proc_tree_schema(proc_tree)
    299     if missing_cols:
--> 300         proc_tree = build_process_tree(procs=proc_tree, schema=schema)
    301 
    302     if schema is None:

~/.local/share/virtualenvs/macOS-ATTACK-DATASET-JrS6VCr6/lib/python3.8/site-packages/msticpy/sectools/proc_tree_builder.py in build_process_tree(procs, schema, show_summary, debug)
    254     elif schema == MDE_EVENT_SCH:
    255         extr_proc_tree = mde.extract_process_tree(procs, debug=debug)
--> 256     merged_procs_keys = _add_tree_properties(extr_proc_tree)
    257 
    258     # Build process paths

UnboundLocalError: local variable 'extr_proc_tree' referenced before assignment
```

**Expected behavior**
Expect a plot of processes with parent and child relationships according to example images

**Screenshots**

**Desktop (please complete the following information):**
 - OS: macOS
 - Browser Chrome
 - Version 93

**Smartphone (please complete the following information):**

**Additional context**

**Comment by ianhelle** [comment]
Hi Jamie,
Thanks for reporting this. Looks like I inadvertently removed the default path when none of the built-in schemas are used.
Apologies. I've found a couple of other issues as well. Hope to get a fix out today.

**Comment by ianhelle** [comment]
This should be fixed in 1.4.1, which is heading to PyPI as I write (referenced PR is above). I also found a few related issues - so thanks for prompting me to look a bit harder.
Let me know if this works/doesn't. Would be happy to work through any problems with you - especially if you have sample data (no or scrubbed PII if you're going to upload here).
Thx

**Comment by ianhelle** [comment]
@jaimeatwork - please re-open this issue if it isn't fixed for you
